Sdram loading

tianyouw opened the pull request:
Adds working linker script, SDRAM test, and LED blink/counter.

tianyouw committed:
Fix HexTool load issue

tianyouw committed:
Fix linker for Dhrystone, change blink demo, add GPIO out for Dhrystone result

tianyouw committed:
Add memory test, increase heap size

Denton-L commented on resources/subproject.mk:
```diff
@@ -5,7 +5,7 @@ ifeq ($(DEBUG),yes)
 
 ifeq ($(DEBUG),no)
-	CFLAGS += -g -O3 
+	CFLAGS += -g -O2 
```
We should perhaps use O0 for something like this

Jiangyi commented on projects/briey/dhrystone/src/main.c:
```diff
@@ -7,6 +7,9 @@
 
 int main() {
+	GPIO_A_BASE->OUTPUT_ENABLE = 0xFFFFFFFF;
+    GPIO_A_BASE->OUTPUT = 0x00000001;
```
Tabbing

tianyouw committed:
Change memtest to compile with debug settings, modify memtest to add 2 states

tianyouw committed:
Refactor memtest

tianyouw committed:
Fix currByte ambiguity in memtest, add NO_OPT setting

Jiangyi committed:
Fix memtest read code

Inverting the bits of a uint8_t always yields a signed int as a
result, and that needs to be casted to an unsigned int for the
comparison to be valid.

tianyouw committed:
Add data cache flush before read back

tianyouw committed:
Minor formating

tianyouw committed:
Clean up stuff

tianyouw committed:
Fix memtest LEDs
